server/container_create_linux.go

```diff
@@ -858,6 +852,77 @@ func setContainerConfigSecurityContext(containerConfig *types.ContainerConfig) {
 	}
 }
 
+// This function returns information from the image, coming from the storage image server.
+func (s *Server) getInfoFromImage(userRequestedImage string) (someNameOfTheImage *references.RegistryImageReference, imageID storage.StorageImageID, someRepoDigest string, imageAnnotations map[string]string, err error) {
+	var imgResult *storage.ImageResult
+	if id := s.StorageImageServer().HeuristicallyTryResolvingStringAsIDPrefix(userRequestedImage); id != nil {
+		imgResult, err = s.StorageImageServer().ImageStatusByID(s.config.SystemContext, *id)
+		if err != nil {
+			return nil, storage.StorageImageID{}, "", nil, err
+		}
+	} else {
+		var potentialMatches []references.RegistryImageReference
+		potentialMatches, err = s.StorageImageServer().CandidatesForPotentiallyShortImageName(s.config.SystemContext, userRequestedImage)
+		if err != nil {
+			return nil, storage.StorageImageID{}, "", nil, err
+		}
+		var imgResultErr error
+		for _, name := range potentialMatches {
+			imgResult, imgResultErr = s.StorageImageServer().ImageStatusByName(s.config.SystemContext, name)
+			if imgResultErr == nil {
+				break
+			}
+		}
+		if imgResultErr != nil {
+			return nil, storage.StorageImageID{}, "", nil, imgResultErr
+		}
+	}
+	// At this point we know userRequestedImage is not empty; "" is accepted by neither HeuristicallyTryResolvingStringAsIDPrefix
+	// nor CandidatesForPotentiallyShortImageName. Just to be sure:
+	if userRequestedImage == "" {
+		err = errors.New("internal error: successfully found an image, but userRequestedImage is empty")
+		return nil, storage.StorageImageID{}, "", nil, err
+	}
+
+	someRepoDigest = ""
+	if len(imgResult.RepoDigests) > 0 {
+		someRepoDigest = imgResult.RepoDigests[0]
+	}
+	return imgResult.SomeNameOfThisImage, imgResult.ID, someRepoDigest, imgResult.Annotations, nil
+}
+
+func (s *Server) getInfoFromArtifact(ctx context.Context, userRequestedImage string) (imageName *references.RegistryImageReference, imageID storage.StorageImageID, someRepoDigest string, imageAnnotations map[string]string, err error) {
+	artifact := ociartifact.New()
+	pullOpts := &ociartifact.PullOptions{}
+
+	var imgManifest manifest.Manifest
+	imgManifest, err = artifact.GetManifest(ctx, userRequestedImage, pullOpts)
+	if err != nil {
+		return nil, storage.StorageImageID{}, "", nil, err
+	}
+
+	var name references.RegistryImageReference
+	name, err = references.ParseRegistryImageReferenceFromOutOfProcessData(userRequestedImage)
```

Comment:
Why do “container images” support image IDs, image ID prefixes, and short name lookups, and “artifacts” (however that differs from “container images”) don’t?

Are the users OK with that?